conda/models/dist.py

```diff
@@ -130,3 +130,18 @@ def __eq__(self, other):
 
     def __ne__(self, other):
         return not self.__eq__(other)
+
+    # ############ conda-build compatibility ################
+
+    def split(self, sep=None, maxsplit=-1):
+        assert sep == '::'
+        return [self.channel, self.dist_name] if self.channel else [self.dist_name]
+
+    def rsplit(self, sep=None, maxsplit=-1):
+        assert sep == '-'
+        assert maxsplit == 2
+        name = '%s::%s' % (self.channel, self.quad[0]) if self.channel else self.quad[0]
+        return name, self.quad[1], self.quad[2]
+
+    def __contains__(self, item):
+        return ensure_text_type(item) in self.__str__()
```

Comment:
If there's much additional code work in this PR, it should likely happen right here.